medusa/providers/generic_provider.py

```diff
@@ -480,6 +482,31 @@ def search(self, search_params, age=0, ep_obj=None):
         """Search the provider."""
         return []
 
+    @staticmethod
+    def parse_pubdate(pubdate, human_time=False, tz=None):
```

Comment:
If we really want to make this more generic, does this need to be a part of providers?  We could move it out of providers altogether and just call it parse_date